Dockerfile

```diff
@@ -13,30 +13,43 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-# Latest Ubuntu LTS
+# If you are building for a foreign target and you get segfaults, try the latest version of qemu
+# $ docker pull tonistiigi/binfmt:latest
+# $ docker run --privileged --rm tonistiigi/binfmt --uninstall qemu-*
+# $ docker run --privileged --rm tonistiigi/binfmt --install arm64
+
 FROM ubuntu:22.04
 ENV DEBIAN_FRONTEND noninteractive
 
-RUN apt-get update && apt-get upgrade -y && apt-get install -y wget nano rsync curl gnupg2 jq unzip bzip2
+# These are automatically provided by docker (no need for --build-arg)
+ARG TARGETPLATFORM
+ARG TARGETARCH
+
+RUN apt-get update && apt-get upgrade -y && apt-get install -y wget nano rsync curl gnupg2 jq unzip bzip2 xz-utils
 
 # for clang-*-15, see https://apt.llvm.org/
 RUN echo "deb http://apt.llvm.org/jammy/ llvm-toolchain-jammy-18 main" >> /etc/apt/sources.list && \
     echo "deb-src http://apt.llvm.org/jammy/ llvm-toolchain-jammy-18 main" >> /etc/apt/sources.list && \
     wget -O - https://apt.llvm.org/llvm-snapshot.gpg.key | apt-key add -
 
 # Install gcc8-arm-none-eabi
-RUN mkdir ~/Downloads &&\
-    cd ~/Downloads &&\
-    wget -O gcc.tar.bz2 https://developer.arm.com/-/media/Files/downloads/gnu-rm/8-2018q4/gcc-arm-none-eabi-8-2018-q4-major-linux.tar.bz2?revision=d830f9dd-cd4f-406d-8672-cca9210dd220?product=GNU%20Arm%20Embedded%20Toolchain,64-bit,,Linux,8-2018-q4-major &&\
-    echo "fb31fbdfe08406ece43eef5df623c0b2deb8b53e405e2c878300f7a1f303ee52 gcc.tar.bz2" | sha256sum -c &&\
-    cd ~/Downloads &&\
-    tar -xjvf gcc.tar.bz2 &&\
-    rm -f gcc.tar.bz2 &&\
-    cd ~/Downloads && rsync -a gcc-arm-none-eabi-8-2018-q4-major/ /usr/local/
+RUN if [ "${TARGETPLATFORM}" = "linux/arm64" ]; then \
+      GNU_TOOLCHAIN=https://developer.arm.com/-/media/Files/downloads/gnu/13.3.rel1/binrel/arm-gnu-toolchain-13.3.rel1-aarch64-arm-none-eabi.tar.xz \
+      GNU_TOOLCHAIN_HASH=c8824bffd057afce2259f7618254e840715f33523a3d4e4294f471208f976764 \
+      GNU_TOOLCHAIN_FORMAT=xz; \
+    else \
+      GNU_TOOLCHAIN=https://developer.arm.com/-/media/Files/downloads/gnu-rm/8-2018q4/gcc-arm-none-eabi-8-2018-q4-major-linux.tar.bz2 \
+      GNU_TOOLCHAIN_HASH=fb31fbdfe08406ece43eef5df623c0b2deb8b53e405e2c878300f7a1f303ee52 \
+      GNU_TOOLCHAIN_FORMAT=bz2; \
```

Comment:
Why not 13 here too? Or will you fix it when rebasing https://github.com/BitBoxSwiss/bitbox02-firmware/pull/1273?

Reply:
I didn't want to update the toolchain in this PR, since it is unrelated. (the gcc-8 toolchain doesn't exist for arm64, so I had to pick something newer)